api/src/main/java/org/geysermc/geyser/api/command/CommandSource.java

```diff
@@ -64,6 +68,17 @@ default void sendMessage(String[] messages) {
      */
     boolean isConsole();
 
+    /**
+     * @return an Optional containing a Java UUID if this source represents any player, otherwise empty
+     */
+    Optional<UUID> playerUuid();
+
+    /**
+     * @return an Optional with a present Connection if this source represents a Bedrock player that is connected
+     * to this Geyser instance. Otherwise, returns an empty optional.
+     */
+    Optional<? extends Connection> connection();
```

Comment:
I'm indifferent

Reply:
An optional is imo way nicer to use, since it directly implies that a connection may not be present

Reply:
Considering the rest of the API does not currently use Optional, I would not use it here. Needs to remain consistent across the board.

Reply:
> An optional is imo way nicer to use, since it directly implies that a connection may not be present

yeah I agree with this, since its easier for people to just ignore nullability annotations. I'll keep it consistent though.